### PR TITLE
[MB-8787] Auto reweigh request check when prime updates shipment actual weight

### PR DESCRIPTION
**Pull request by duncan-truss**
## Description

This PR adds functionality that auto requests a reweigh for all shipments of a move if the actual weight is updated and the sum reaches 90% of the weight allowance.

It assumes the shipment has already been approved (not cancelled or a diversion kicked back to submitted) and visible on the MTO page.

The weight sum will take the lower of the actual weight and reweigh weight for a shipment (of the shipment being updated or another shipment on the move) when doing the calculation.

## Reviewer Notes

Should we modify the Prime shipment response payloads to include the reweigh object?  It seems to not be included but maybe that work is being done elsewhere?

There is no hard constraint that the actual weight has to be updated before the reweigh weight to my knowledge but I'm not sure that would happen in practice?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

1. Login as the TOO user
2. Locate the move with locator `INXSWT` and select it from the queue
3. Approve both shipments on the move with a basic service item making it available to the prime.
4. Either from the dev console or from the prime API grab one of the shipment ids from the move.
5. Make a PATCH request using the `updateMTOShipment` prime endpoint, setting the mtoShipmentID var in the path and the If-Match header.  The PATCH request JSON body should set the PrimeActualWeight to 90% of the weight allowance (8,000 lbs)
6. Navigate to the Move Task Order page and observe the shipment with the actual weight of 7200 displayed and tags that both shipments now have reweigh requests

Screen Shot 2021-09-03 at 11 27 11 AM

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-8787) for this change

## Screenshots

Screen Shot 2021-09-03 at 10 03 00 AM
Screen Shot 2021-09-03 at 10 03 11 AM

### CODE REVIEW

**Comment by duncan-truss** [comment]
I'm still working on adding tests for the mto_shipment_updater service that the prime handler calls

File: pkg/services/mto_shipment/mto_shipment_updater_test.go [hunk 1]
```diff
@@ -1148,3 +1148,101 @@ func (suite *MTOShipmentServiceSuite) TestUpdateShipmentEstimatedWeightMoveExces
...
+		err = suite.DB().Eager("Reweigh").Reload(&primeShipment)
+		suite.NoError(err)
+
+		suite.NotNil(primeShipment.Reweigh)
+		suite.Equal(primeShipment.ID.String(), primeShipment.Reweigh.ShipmentID.String())
+		suite.NotNil(primeShipment.Reweigh.RequestedAt)
+		suite.Equal(models.ReweighRequesterSystem, primeShipment.Reweigh.RequestedBy)
+	})
+
+	suite.T().Run("Skips calling check auto reweigh if actual weight was not provided in request", func(t *testing.T) {
```
Comment (Ryan-Koch) [inline pkg/services/mto_shipment/mto_shipment_updater_test.go:1192]: I appreciate testing this case, definitely a trip wire one could accidentally hit making changes to update the object later on.

File: pkg/services/move/move_weights_test.go [hunk 2]
```diff
@@ -236,3 +241,225 @@ func (suite *MoveServiceSuite) TestExcessWeight() {
...
+		suite.NoError(err)
+
+		err = suite.DB().Eager("Reweigh").Reload(&existingShipment)
+		suite.NoError(err)
+		suite.Equal(uuid.Nil, existingShipment.Reweigh.ID)
+		suite.Equal(uuid.Nil, approvedShipment.Reweigh.ID)
+		mockedReweighRequestor.AssertNotCalled(suite.T(), "RequestShipmentReweigh")
+	})
+
+	/*
```
Comment (Ryan-Koch) [inline pkg/services/move/move_weights_test.go:440]: Are these test cases ones we're hoping to use eventually or should they go?

**Review by Ryan-Koch — APPROVED** [review]

File: pkg/services/move/move_weights_test.go (hunk 2 above)
Comment (duncan-truss) [reply pkg/services/move/move_weights_test.go:440]: Thank you I forgot about those, let me update them.

**Pull request merged**